lib/AST/GraphOperationInfo.cpp

```diff
@@ -193,3 +196,16 @@ int tf::decodeShapeAttr(const ASTContext &ctx, SymbolicValue attr,
   }
   return arrayValue.size();
 }
+
+/// Return the TF_DataType represented by `value`.
+unsigned tf::getTFDataType(SymbolicValue value) {
+  value = value.lookThroughSingleElementAggregates();
+  assert(value.getKind() == SymbolicValue::Integer);
+  return value.getIntegerValue().sextOrTrunc(32).getLimitedValue();
+}
+
+/// Return a SymbolicValue that represents the TF_DataType for the given swift
+/// type.
+SymbolicValue tf::convertSwiftTypeToTFSymbolicValue(Type type) {
+  return SymbolicValue::getInteger(convertSwiftTypeToTF(type), 32);
```

Comment:
does the input type must be valid? 

if so it'd be good to assert convertSwiftTypeToTF(type) does not return 0 (or other invalid value).

Reply:
yep, will assert